src/main/java/org/yaml/snakeyaml/introspector/PropertyUtils.java

```diff
@@ -57,6 +52,13 @@ public PropertyUtils() {
     if (platformFeatureDetector.isRunningOnAndroid()) {
       beanAccess = BeanAccess.FIELD;
     }
+
+    /* when running with jlink restricted JDK without java.desktop package */
+    try {
```

Comment:
Could you move this detection into the `PlatformFeatureDetector` maybe in something like `isIntrospectorAvailable` ?

Reply:
Done in 2ca1828. Is that OK?